<?xml version="1.0"?>

<launch>
    
    <arg name="tag_visualize" default="false" />
    <arg name="rviz" default="true" />
    <arg name="tune" default="false" />
    <arg name="tune_x" default="false" />
    <arg name="tune_y" default="false" />
    <arg name="tune_yaw" default="false" />

    <rosparam command="load" file="$(find wp_nav_pid)/params/wp_nav_pid.yaml" />
    <rosparam command="load" file="$(find user_params)/params/chassis_config.yaml" />

    <include file="$(find car_localization)/launch/car_localization.launch" >
        <arg name="visualize" value="$(arg tag_visualize)" />
        <arg name="rviz" value="$(arg rviz)" />
    </include>
    
    <node pkg="wp_nav_pid" type="wp_nav_pid_node.py" name="wp_nav" output="screen" >
        <param name="tune" value="$(arg tune)" />
        <param name="tune_x" value="$(arg tune_x)" />
        <param name="tune_y" value="$(arg tune_y)" />
        <param name="tune_yaw" value="$(arg tune_yaw)" />
    </node>

    <node pkg="chassis" name="chassis" type="chassis_control_node.py" output="screen" />

    <group if="$(arg tune_x)">
        <group ns="pid_x">
            <node pkg="wp_nav_pid" type="pid_tuner.py" name="pid_tuner_x" output="screen" />
        </group>
    </group>

    <group if="$(arg tune_y)">
        <group ns="pid_y">
            <node pkg="wp_nav_pid" type="pid_tuner.py" name="pid_tuner_y" output="screen" />
        </group>
    </group>

    <group if="$(arg tune_yaw)">
        <group ns="pid_yaw">
            <node pkg="wp_nav_pid" type="pid_tuner.py" name="pid_tuner_yaw" output="screen" />
        </group>
    </group>

</launch>
